typegen/trace_data_file_collector.py

```diff
@@ -12,12 +12,12 @@
 class DataFileCollector(ABC):
     """Collects data files in a given path."""
 
-    def __init__(self):
-        self.file_pattern: str = ""
-        self.collected_data = list()
+    def __init__(self, file_pattern: str):
+        self.file_pattern = file_pattern
```

Comment:
Do not use file_pattern as an argument if every instance of a subclass has the same file pattern. If that is the case, define a protected abstract _get_file_pattern() -> str whose sub classes return their own file pattern constant and drop the file_pattern class member.

Reply:
Your recommendation will actually consume more memory because we will be creating a new string for each call. It's not like we're going to have thousands of these flying around at once. The memory consumption for this is negligible.

Reply:
The proposal is not about memory consumption, but about readability. This is actually how you would override properties. If you require a unique constant for each class, but each subclass has a different "constant", then define a abstract property, which will be overridden by each subclass. Since the return value is a string, the creation of a new str each time is minimal regarding memory consumption. However, this usage of a property means that you do not read the file pattern parameter anymore. 

Unless we need instances whose file pattern differ from other instances of the same class, please change this.

Reply:
You know what, it does not matter. Your implementation is also alright.